src/components/Clock.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class Clock extends React.Component {
+  static propTypes = {
+    name: PropTypes.number.isRequired,
+  }
+
+  state= {
+    time: new Date(),
+  };
+
+  componentDidMount() {
+    this.timeId = setInterval(() => {
+      const date = new Date();
+
+      this.setState({ time: date });
+
+      // eslint-disable-next-line
+      console.log(date.toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { name } = this.props;
+
+    if (name === prevProps.name) {
+      return;
+    }
+
+    // eslint-disable-next-line
+    console.log(`The Clock was renamed from ${prevProps.name} to ${name}`);
+  }
```

Comment:
It's good that you have used componentDidUpdate, but for this task, you just need only to log to the console new name after changing it. So you can simply put your console.log into your 'changeName' method